src/WpfMath/TexFormulaHelper.cs

```diff
@@ -153,9 +153,34 @@ public void AddOperator(string operatorFormula, string lowerLimitFormula, string
                 formulaParser.Parse(upperLimitFormula), useVerticalLimits);
         }
 
+        public int MoveOffset(Atom atom, int pos)
+        {
+            if (atom.Source != null)
+            {
+                atom.Source = new SourceSpan(atom.Source.Source, pos, atom.Source.Length);
```

Comment:
Please leave a thorough comment about what're you doing here. Every piece of code that deals with sources in this way (rebasing sources to other place in formula) demands a detailed comment on what's going on.